packages/nimble-components/src/table/specs/README.md

```diff
@@ -132,13 +132,17 @@ The various APIs/features of the `nimble-table` will be split up amongst several
     -   Define how the action menu gets associated with a particular column
     -   Define the table-level(column-level?) API(s) for applying an action menu to the table (slot, properties, etc...)
 
+_Attributes_
+
+-   `id-field-name` - An optional string attribute that specifies the field name within a row's record to use as a row's ID. If the attribute is not defined or a field with the specified name does not exist in a row's record, a default ID will be created. The default ID is the index of the row's record within the table's `data` property. The value associated with the field name must be unique across all records in the table's data.
```

Comment:
What would happen if the values of the field are not unique across rows? I guess we could document that they must be unique, but if clients don't obey that, would the table generate a useful error or would we end up with weird behaviors?

Reply:
These are similar to questions asked below, so I am consolidating the discussion.

https://github.com/ni/nimble/pull/937#discussion_r1061055541